File: 0501_0600/0543/0543.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n, w, d, p int
	fmt.Fscan(f, &n, &w, &d, &p)
	s := (w*n*(n-1)/2 - p) / d
	if s == 0 {
		s = n
	}
	fmt.Fprint(q, s)
}

File: 0501_0600/0543/0543.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
	n, w, d, p = (int(x) for x in f.read().split())
	s = (w * n * (n - 1) // 2 - p) // d
	q.write(str(s if s else n))
